### PR TITLE
Generate validation report as JSON

### LINKED ISSUE #21: Validator report in json format
Actually we have the following behaviour

````
spid-compliant-certificates validator --crt-file src/spid_sp_test/idp/public.cert 
...

[INFO ] Validating certificate src/spid_sp_test/idp/public.cert against public sector specifications
[INFO ] Checking the key type and size: success
[INFO ]     The keypair must be RSA
[INFO ]     The key size must be greater than or equal to 2048 (now: 2048)
[INFO ]     The key size must be one of [2048, 3072, 4096] (now: 2048)
[INFO ] Checking the signature digest algorithm: success
[INFO ]     The digest algorithm must be one of ['sha256', 'sha512'] (now: sha256)
[ERROR] Checking the SubjectDN: failure
[INFO ]     Name attribute [{attr._name}, {attr.dotted_string}] is not allowed in subjectDN
[INFO ]     Name attribute [{attr._name}, {attr.dotted_string}] is not allowed in subjectDN
[INFO ]     Name attribute [{attr._name}, {attr.dotted_string}] is not allowed in subjectDN
[INFO ]     Name attribute [{attr._name}, {attr.dotted_string}] is not allowed in subjectDN
[INFO ]     Name attribute [{attr._name}, {attr.dotted_string}] is not allowed in subjectDN
[INFO ]     Name attribute [{attr._name}, {attr.dotted_string}] is not allowed in subjectDN
[ERROR]     Name attribute [organizationIdentifier, 2.5.4.97] must be present in subjectDN
[ERROR]     Name attribute [uri, 2.5.4.83] must be present in subjectDN
[INFO ]     Name attribute [commonName, 2.5.4.3] must be present in subjectDN
[INFO ]     Name attribute [countryName, 2.5.4.6] must be present in subjectDN
[INFO ]     Name attribute [localityName, 2.5.4.7] must be present in subjectDN
[INFO ]     Name attribute [localityName, 2.5.4.7] must be present in subjectDN
[INFO ]     Name attribute [organizationName, 2.5.4.10] must be present in subjectDN
[INFO ]     Value for name attribute [countryName, 2.5.4.6] can not be empty
[INFO ]     Value for name attribute [countryName, 2.5.4.6] is not a valid country code (IT)
[INFO ]     Value for name attribute [stateOrProvinceName, 2.5.4.8] can not be empty
[INFO ]     Value for name attribute [localityName, 2.5.4.7] can not be empty
[INFO ]     Value for name attribute [organizationName, 2.5.4.10] can not be empty
[INFO ]     Value for name attribute [organizationalUnitName, 2.5.4.11] can not be empty
[INFO ]     Value for name attribute [commonName, 2.5.4.3] can not be empty
[ERROR] Checking basicConstraints x509 extension: failure
[INFO ]     basicConstraints can not be set as critical
[ERROR]     CA must be FALSE
[ERROR] Checking keyUsage x509 extension: failure
[ERROR]     keyUsage must be present
[ERROR] Checking certificatePolicies x509 extension: failure
[ERROR]     certificatePolicies must be present
[ERROR] The certificate src/spid_sp_test/idp/public.cert does not match public sector specifications
````

That's awesome but probably there woudl also the need to have a report in a json format.
This could be implemented here:
https://github.com/italia/spid-compliant-certificates-python/blob/ba7209f889252503e2f182ba2007880ee26004a1/spid_compliant_certificates/validator/validate.py#L40

instead of log the message in stdout we could handle it with an abstract Report class that would be initialized with the argparse option. 

At the same way we could define many python logger handlers and select one of them using argparse CLI. Does this fit to you?

### LINKED ISSUE #22: Wrong f-string syntax
**Describe the bug**

The `f` at the beginning is missing.

https://github.com/italia/spid-compliant-certificates-python/blob/bfb32d10430d7fb78802c803a306c9804fb013a2/spid_compliant_certificates/validator/checks/subject_dn.py#L63

**To Reproduce**

None

**Expected behavior**

It should be `f'something'`

**Screenshots**

```
[...]
[E]   Checking the SubjectDN
[I]     Name attribute [{attr._name}, {attr.dotted_string}] is not allowed in subjectDN
[I]     Name attribute [{attr._name}, {attr.dotted_string}] is not allowed in subjectDN
[I]     Name attribute [{attr._name}, {attr.dotted_string}] is not allowed in subjectDN
[I]     Name attribute [{attr._name}, {attr.dotted_string}] is not allowed in subjectDN
[I]     Name attribute [{attr._name}, {attr.dotted_string}] is not allowed in subjectDN
[I]     Name attribute [{attr._name}, {attr.dotted_string}] is not allowed in subjectDN
[...]
```

**Environment (please complete the following information):**

None

**Additional context**

None

### PR DESCRIPTION
**Pull request by psmiraglia**
**Scope of the pull request**

The PR introduces the `--out-form` and `--out-file` switches. They instruct the application about **how** the validation report must be serialized (e.g. JSON, TXT, XML) and **where** it must be saved.

```
$ spid-compliant-certificates validator --out-file report.json --out-form json
   _____ _____ _____ _____     _____                      _ _             _
  / ____|  __ \_   _|  __ \   / ____|                    | (_)           | |
 | (___ | |__) || | | |  | | | |     ___  _ __ ___  _ __ | |_  __ _ _ __ | |_
  \___ \|  ___/ | | | |  | | | |    / _ \| '_ ` _ \| '_ \| | |/ _` | '_ \| __|
  ____) | |    _| |_| |__| | | |___| (_) | | | | | | |_) | | | (_| | | | | |_
 |_____/|_|   |_____|_____/   \_____\___/|_| |_| |_| .__/|_|_|\__,_|_| |_|\__|
   _____          _   _  __ _           _          | |
  / ____|        | | (_)/ _(_)         | |         |_|
 | |     ___ _ __| |_ _| |_ _  ___ __ _| |_ ___  ___
 | |    / _ \ '__| __| |  _| |/ __/ _` | __/ _ \/ __|
 | |___|  __/ |  | |_| | | | | (_| (_| | ||  __/\__ \
  \_____\___|_|   \__|_|_| |_|\___\__,_|\__\___||___/  v0.1




[I] Validating certificate /path/to/crt.pem against public sector specifications
[I] Certificate /path/to/crt.pem matches the public sector specifications

[ . . . ]

[I] Saving report as JSON in /path/to/report.json
```

**Closed issues**

- Close #21 
- Close #22

**Added issues**

- #24
- #25

**Tests**

- [X] I manually tested it
- [ ] I added unit test
- [ ] I ran the existing unit test and they do not fail

### CODE REVIEW

File: spid_compliant_certificates/validator/validate.py [hunk 1]
```diff
@@ -70,43 +43,42 @@ def validate(crt_file: str, sector: str) -> None:
...
     # check key type and size
-    results.append(_do_check(
+    r.add_test(_do_check(
         checks.key_type_and_size(crt),
         'Checking the key type and size'
     ))
 
     # check digest algorithm
-    results.append(_do_check(
+    r.add_test(_do_check(
```
Comment (peppelinux) [inline spid_compliant_certificates/validator/validate.py]: r collides with pdb return command
https://docs.python.org/3/library/pdb.html#pdbcommand-return

better use something different

**Review by peppelinux — CHANGES_REQUESTED** [review]

File: spid_compliant_certificates/validator/validate.py (hunk 1 above)
Comment (psmiraglia) [reply spid_compliant_certificates/validator/validate.py]: Really? Are you saying `pdb` can't be used if your code defines a variable named `r`? Anyway, I'll change it to `rep.add_test()`.

Comment (peppelinux) [reply spid_compliant_certificates/validator/validate.py]: ... then it would need some trick like `!r` but better don't collide :) thank you

Comment (psmiraglia) [reply spid_compliant_certificates/validator/validate.py]: If everything is ok, could you please resolve the conversation?

**Review by peppelinux — APPROVED** [review]

**Pull request merged**